### PR TITLE
Context for authentification

### PR DESCRIPTION
**Pull request by fuentesloic**

### CODE REVIEW

File: App.js
Comment (vinyll) [inline App.js]: Wondering about holding this in _static/context_

File: static/context.js
Comment (vinyll) [inline static/context.js]: Well, this _context.js_ script is a pure system code that don't seem to have its place in _static_ at all.
Wondering what made you opt for this?

File: static/context.js
Comment (vinyll) [inline static/context.js]: You'll really need to explain to me some day why we need to bind the current scope to the `api` function 🤔

File: static/context.js
Comment (vinyll) [inline static/context.js]: I'm not sure how `Context` works.
It feels misplaced to have the `login()` function here intead of holding it inside the `AuthScreen`.
It feels like leaving the _component_ approach as this depends on the AuthScreen and the other way around.

Can't the AuthScreen send the user infos to the _Context_?

File: static/context.js
Comment (fuentesloic) [reply static/context.js]: Yep so same level than the App.js

File: static/context.js
Comment (fuentesloic) [reply static/context.js]: That was just a test (and it's not working ^^)

File: static/context.js
Comment (fuentesloic) [reply static/context.js]: yes sure and now I think it's a better idea, I was just trying ... and that was not working so ...

**Comment by fuentesloic** [comment]
@vinyll ready for merge in master, could you be review ? could you be reviewwwwww ^^ [could you be love, could you be loveeeeee](https://www.youtube.com/watch?v=Aj-jZsACbtg)

File: components/Buttons/CommunityButton.js
Comment (vinyll) [inline components/Buttons/CommunityButton.js]: Why not moving this into the `render()` method?

```js
const title = this.props.communityMode
    ? 'Sharing location with community'
    : 'Stealth mode – not socializing'
return (
    <CheckBox…
```

File: components/screens/AppLoadingScreen.js
Comment (vinyll) [inline components/screens/AppLoadingScreen.js]: Did you mean `updateValue('userId', null)`?

File: components/screens/AppScreen.js
Comment (vinyll) [inline components/screens/AppScreen.js]: This looks like a dev message for debugging purpose, you probably wanted to remove this.

File: context.js
Comment (vinyll) [inline context.js]: It feels like you are redifining `setState()`.
What is the purpose of this?
`updateValue` seems to be passed all along; I'm wondering if passing the `setState` would not be more straighforward?
If there's an actual need of saying `updateValue(key, value)` instead of `setState({key: value})`, maybe you could then call it `setStateValue(key, value)`. But I still question why redefining a method for that.

File: context.js
Comment (vinyll) [inline context.js]: maybe you could say here—not to confuse with local state:

```js
    …,
    setGlobalState: this.setState
```

and later call `setGlobalState({userId: '…'})` and `setGlobalState({communityMode: true})` when using?

File: components/screens/AppLoadingScreen.js
Comment (fuentesloic) [reply components/screens/AppLoadingScreen.js]: better indeed

File: components/screens/AppScreen.js
Comment (fuentesloic) [reply components/screens/AppScreen.js]: also true

File: components/Buttons/CommunityButton.js
Comment (vinyll) [inline components/Buttons/CommunityButton.js]: [unindent](https://www.youtube.com/watch?v=Zf8ir_gvbwA) 🎶

**Review by vinyll — CHANGES_REQUESTED** [review]

File: components/Header/HeaderApp.js
Comment (vinyll) [inline components/Header/HeaderApp.js]:
```js
const avatar = this.props.userId !== null && <AvatarApp userId={this.props.userId} />
```

File: context.js
Comment (fuentesloic) [reply context.js]: I neet to transmit a method to update the context state and so the provider is up to date and dispatch towards the consumer the state evolution

File: components/Header/HeaderApp.js
Comment (fuentesloic) [reply components/Header/HeaderApp.js]: :( done

File: context.js
Comment (fuentesloic) [reply context.js]: setGlobalState is more accurate :)

File: components/Buttons/CommunityButton.js
Comment (fuentesloic) [reply components/Buttons/CommunityButton.js]: is it on my Atom (even on VScode) ... any clues ??

File: components/Header/HeaderApp.js
Comment (vinyll) [inline components/Header/HeaderApp.js]: unindent

File: context.js
Comment (vinyll) [inline context.js:20]: If it were me, I would totally drop this function.
You say it is necessary to keep it for some reason I'm not sure of.
If it's the case you could write:

```js
setGlobalState (obj) {
  this.setState(obj)
}
```

And then use it just like `setState()`: `setGlobalState({userId: null})`

File: context.js
Comment (vinyll) [inline context.js:34]: Did you test writing `setGlobalState: this.setState` and dropping the intermediary function?

Comment (fuentesloic) [reply context.js:34]: in first time I wrote `setGlobalState: this.setState` but that was not working, I tried different ways like giving the `this` and use in component `this.setState` but same effect :(

File: context.js
Comment (fuentesloic) [reply context.js:20]: I agree with you but I just tried with: 
```
setGlobalState (obj) {
  this.setState(obj)
}
```

And it's not working, I guess we are in a limitation of how react can pass the setState function you can test it with this commit e56ae153e44c22bd6f54b4a2123c9c738dc040f0

File: components/Header/HeaderApp.js
Comment (fuentesloic) [reply components/Header/HeaderApp.js]: Unindent fix (my tab in VScode at job made this ... not my fault 👼 )
About `const avatar` it can't be false so I have to write this ternaire 2554718f746b05e9e074912e89eeb60f8963f537

File: context.js
Comment (vinyll) [reply context.js:34]: Ok then, technical limitation convinced me.
I leave it to your opinion about whether write `setGlobalState(key, value)` or `setGlobalState({key: value})`

**Review by vinyll — APPROVED** [review]
With a small notice.
Cool it works!! :D

File: context.js
Comment (fuentesloic) [reply context.js:34]: @vinyll put in issue #12

**Pull request merged**
